table-api/src/test/java/io/deephaven/api/filter/FilterComparisionTest.java

```diff
@@ -0,0 +1,210 @@
+/**
+ * Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+ */
+package io.deephaven.api.filter;
+
+import io.deephaven.api.ColumnName;
+import io.deephaven.api.RawString;
+import io.deephaven.api.Strings;
+import io.deephaven.api.filter.FilterComparison.Operator;
+import io.deephaven.api.literal.Literal;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class FilterComparisionTest {
+
+    static final ColumnName FOO = ColumnName.of("Foo");
+    static final ColumnName BAR = ColumnName.of("Bar");
+    static final Literal V42 = Literal.of(42L);
+    static final RawString E42 = RawString.of("41 + 1");
+
+    static final FilterComparison FOO_EQ_42 = FilterComparison.eq(FOO, V42);
+    static final FilterComparison FOO_GT_42 = FilterComparison.gt(FOO, V42);
+    static final FilterComparison FOO_GTE_42 = FilterComparison.gte(FOO, V42);
+    static final FilterComparison FOO_LT_42 = FilterComparison.lt(FOO, V42);
+    static final FilterComparison FOO_LTE_42 = FilterComparison.lte(FOO, V42);
+    static final FilterComparison FOO_NEQ_42 = FilterComparison.neq(FOO, V42);
+
+    static final FilterComparison FOO_EQ_BAR = FilterComparison.eq(FOO, BAR);
+    static final FilterComparison FOO_GT_BAR = FilterComparison.gt(FOO, BAR);
+    static final FilterComparison FOO_GTE_BAR = FilterComparison.gte(FOO, BAR);
+    static final FilterComparison FOO_LT_BAR = FilterComparison.lt(FOO, BAR);
+    static final FilterComparison FOO_LTE_BAR = FilterComparison.lte(FOO, BAR);
+    static final FilterComparison FOO_NEQ_BAR = FilterComparison.neq(FOO, BAR);
+
+    static final FilterComparison FOO_EQ_E42 = FilterComparison.eq(FOO, E42);
+    static final FilterComparison FOO_GT_E42 = FilterComparison.gt(FOO, E42);
+    static final FilterComparison FOO_GTE_E42 = FilterComparison.gte(FOO, E42);
+    static final FilterComparison FOO_LT_E42 = FilterComparison.lt(FOO, E42);
+    static final FilterComparison FOO_LTE_E42 = FilterComparison.lte(FOO, E42);
+    static final FilterComparison FOO_NEQ_E42 = FilterComparison.neq(FOO, E42);
+
+    @Test
+    void lhs() {
+        assertThat(FOO_EQ_42.lhs()).isEqualTo(FOO);
```

Comment:
I feel somewhat motivated to consolidate the lhs, rhs, operation, and strings test into one helper method, as in

```
checkAllAspects(FOO_EQ_42, FOO, Operator.EQUALS, V42, "(Foo) == (42L)")
checkAllAspects(FOO_GT_42, FOO, Operator.GREATER_THAN, V42, "(Foo) > (42L)")
```

etc etc